Toggle buttons

SammieScully opened the pull request:
Added js functions to buttons to toggle between like and unlike on the specific-post page

iansteyn reviewed (changes requested):
This works great so far, but there's a few things that are still missing and it's not quite re-usable (it needs to be so we can add functionality to buttons in post-summary).

### Missing buttons in Specific post
-  edit button
- save button

### Add functionality for edit
- it should take you to the create post page

### Turn 'toggle' functionality into something that works for all like/save buttons
What you are doing for like/unlike right now works pretty well, although it might be easier to change the stuff inside the button element rather than the whole button - mostly because I want it to be reused for the buttons in post-summary.

Do this by hiding/showing the filled/unfilled versions of the svg and text inside the button.

If you find its easier to stick with what you were already doing before, that's fine - at least just add the save toggle functionality too pls.

> [!TIP]
> Also, note that we have a special class defined in `main.css` called `hidden` - you can just add/remove/toggle that to an elements `classList` to show/hide (it's basically the same as setting style to `display:hidden` but standardized 🤷)

iansteyn commented:
Once you've done that stuff I'm happy to finish making it actually work for the post-summary buttons

SammieScully commented:
Added in save and edit
Save is functional
Also added icons to post and discard
Other than that I stcuk with what I had been doing as Im tired and confused

iansteyn reviewed (changes requested):
OK this looks great. 
✅ Save/unsave works
✅ Like/unlike works
✅ edit works

❗ One last thing: can you pls put them in the same order as post-summary? 
- like, save, edit, delete
